### PR TITLE
UP-4855: Support Gradle Release to Sonatype/Maven Central

### PR DESCRIPTION
**Pull request by ChristianMurphy**
##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]

##### Description of change

https://issues.jasig.org/browse/UP-4855

*edit* original changes were merged out of band to master, this addresses additional comments added to ticket.

- [x] Create new release doc
- [x] Signature Validation
- [x] POM Validation
- [x] Sources Validation
- [x] Javadoc Validation

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

**Comment by ChristianMurphy** [comment]
Could appveyor be restarted for latest commit?
I think the error is a false positive.

```
Failed to execute goal on project uportal-war: Could not resolve dependencies for project org.jasig.portal:uportal-war:war:5.0.0-SNAPSHOT: The following artifacts could not be resolved: org.jasig.portal:uPortal-api-internal:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-api-platform:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-api-platform-impl:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-api-rest:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-api-search:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-core:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-concurrency:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-content-portlet:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-content-publishing:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-events:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-groups-core:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-groups-filesystem:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-groups-grouper:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-groups-ldap:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-groups-local:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-groups-pags:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-groups-smartldap:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-hibernate:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-i18n:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-io-core:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-io-jaxb:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-io-types:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-layout-core:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-layout-impl:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-marketplace:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-persondir:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-portlets:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-rendering:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-rdbm:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-security-authn:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-security-core:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-security-mvc:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-security-permissions:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-security-services:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-security-xslt:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-soffit-connector:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-spring:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-tenants:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-tools:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-url:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-utils-core:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-utils-jmx:jar:5.0.0-SNAPSHOT, org.jasig.portal:uPortal-web:jar:5.0.0-SNAPSHOT: Could not find artifact org.jasig.portal:uPortal-api-internal:jar:5.0.0-SNAPSHOT
```

File: gradle.properties [hunk 1]
```diff
@@ -17,8 +17,7 @@
 # under the License.
 #
 
-# TODO:  Change to org.apereo.portal
-group=org.jasig.portal
+group=org.apereo.portal
```
Comment (ChristianMurphy) [inline gradle.properties]: Is uPortal able to support `org.apereo`?

Comment (drewwills) [inline gradle.properties]: Not yet... not to my knowledge

Comment (ChristianMurphy) [inline gradle.properties]: I'll get that rolled back

**Comment by ChristianMurphy** [comment]
https://blog.codefx.org/tools/snapshots-gradle-maven-publish-plugin/

**Comment by ChristianMurphy** [comment]
Snapshot release should now be uploadable using
``` sh
./gradlew publish -DsonatypeUsername=<UID> -DsonatypePassword=<PASSWORD>
```

**Comment by ChristianMurphy** [comment]
:notebook: Because the number of gradle levels vary in the project, the `publishing.publication` may need to be moved into the individual sub-projects.

**Comment by drewwills** [comment]
@ChristianMurphy -- just tried again.  MUCH closer.

Still 2 validation errors...
  - failureMessage	Invalid POM: /org/jasig/portal/uPortal-content/5.0.0-M1/uPortal-content-5.0.0-M1.pom: Project name missing
  - failureMessage	Missing Signature: '/org/jasig/portal/uPortal-content/5.0.0-M1/uPortal-content-5.0.0-M1-sources.jar.asc' does not exist for 'uPortal-content-5.0.0-M1-sources.jar'.

**Comment by ChristianMurphy** [comment]
For signing https://github.com/ChristianMurphy/uPortal/blob/17c9ae111ff72282f89f327e257f9022b8858aa5/docs/developer/other/RELEASE.md and https://docs.gradle.org/current/userguide/signing_plugin.html#sec:signatory_credentials may provide some insights.

PR has been updated with Name attribute added to generated pom.

File: docs/developer/other/RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+signing.password=secret
+signing.secretKeyRingFile=/Users/me/.gnupg/secring.gpg
+```
+
+## Running a release
+
+Run
+
+```sh
+./gradlew publish -P sonatypeUser=[user] -P sonatypePassword={pass}
```
Comment (apetro) [inline docs/developer/other/RELEASE.md]: Is the duplicate `-P` s correct?

Comment (drewwills) [inline docs/developer/other/RELEASE.md]: I believe it is correct, yes;  and FWIW this part appears to be working well.

**Comment by drewwills** [comment]
@ChristianMurphy et al. -- I just pushed some changes.  Publishing is working now.

**Comment by ChristianMurphy** [comment]
Huzzah! :+1: :100: 
This will need to be updated to make the `ossrhUsername` and `ossrhPassword` optional.
Then this will LGTM. :star2:

**Comment by drewwills** [comment]
Aren't they optional already?  They should net be needed unless you run the `uploadArchives` task.

**Comment by ChristianMurphy** [comment]
```
FAILURE: Build failed with an exception.
* Where:
Build file '/home/travis/build/Jasig/uPortal/build.gradle' line: 111
* What went wrong:
A problem occurred evaluating root project 'uPortal'.
> Could not get unknown property 'ossrhUsername' for object of type org.gradle.api.publication.maven.internal.deployer.DefaultGroovyMavenDeployer.
```

https://travis-ci.org/Jasig/uPortal/builds/239658612#L181-L188
https://ci.appveyor.com/project/drewwills/uportal/build/1.0.383#L95

**Pull request merged**